src/django_components/component_media.py

```diff
@@ -377,14 +379,24 @@ def get_context_data(self):
     # as relative to the directory where the component class is defined.
     _resolve_component_relative_files(comp_cls, comp_media, comp_dirs=comp_dirs)
 
-    # If the component defined `js_file` or `css_file`, instead of `js`/`css` resolve them now.
-    # Effectively, even if the Component class defined `js_file`, at "runtime" the `js` attribute
+    # If the component defined `template_file`, `js_file` or `css_file`, instead of `template`/`js`/`css`,
+    # we resolve them now.
+    # Effectively, even if the Component class defined `js_file` (or others), at "runtime" the `js` attribute
     # will be set to the content of the file.
-    comp_media.js = _get_static_asset(
-        comp_cls, comp_media, inlined_attr="js", file_attr="js_file", comp_dirs=comp_dirs
+    # So users can access `Component.js` even if they defined `Component.js_file`.
+    comp_media.template = _get_asset(
```

Comment:
This here sets the contents of the file set by `Component.template_file` into `Component.template`.